crates/bevy_ecs/src/change_detection.rs

```diff
@@ -28,6 +28,13 @@ pub const MAX_CHANGE_AGE: u32 = u32::MAX - (2 * CHECK_TICK_THRESHOLD - 1);
 /// Normally change detecting is triggered by either [`DerefMut`] or [`AsMut`], however
 /// it can be manually triggered via [`DetectChanges::set_changed`].
 ///
+/// To ensure that changes are only flagged when the value actually differs,
+/// check if the value is equal before assignment.
+/// You must be *sure* that you are not mutably derefencing in this process.
+///
+/// The [`set_if_differs`](DetectChanges::set_if_differs) method
+/// provides a helper method for this common functionality.
```

Comment:
```suggestion
/// [`set_if_differs`](DetectChanges::set_if_differs) is a helper
/// method for this common functionality.
```
"method provides a helper method" sounds odd to me.